src/main.cpp

```diff
@@ -800,6 +800,37 @@ static std::string FormatStateMessage(const CValidationState &state)
         state.GetRejectCode());
 }
 
+bool CheckForConflicts(const CTransaction &tx)
+{
+    LOCK(mempool.cs);
+
+    CCoinsView dummy;
+    CCoinsViewCache view(&dummy);
+
+    CCoinsViewMemPool viewMemPool(pcoinsTip, mempool);
```

Comment:
You could have a ```const CTxMemPool& pool``` parameter instead of using the global mempool (like AcceptToMemoryPool).
The same goes for pcoinsTip.